### ISSUE TITLE
Crash on Linux when NFD is initialized before GLFW window is created

### ISSUE THREAD

**Issue by JamesBoer** [btzy/nativefiledialog-extended#21]
This isn't so much a bug to be fixed, but due to the restructuring of code from NFD requiring explicit initialization, there's now a potential for causing problems for the Linux platform.

The issue is that, for applications using GLFW, if you initialize NFD _before_ you initialize GLFW and create a window, you'll get a mysterious crash deep in GTK code whenever you try to open a native dialog on Linux.  This may be true of other X11-based apps besides GLFW, but I can't confirm this.

This is definitely something you should make note of in your documentation.  I'm just thankful I found the cause, since I appreciate the enhancements you made in this NFD fork.  I was planning to fix the same things you did in the original NFD library before I found your version.  Thanks for the work you've done in enhancing the library!

**Comment by btzy** [comment]
Thanks for reporting this!  I'm glad you found my enhancements useful.

It has been more than two years since I used NFD with SDL2, and I vaguely remember having some issues when trying to display unicode characters in the title bar of the window if I initialize NFD before initializing SDL.  I didn't investigate much into it, but I presumed that SDL assumes nothing has been initialized yet and initializes everything to the way SDL likes it to be (in which case it would arguably be SDL's bug).

This might be worth investigating a bit, though I think we'll probably find that the "proper" fix is in GLFW/SDL instead of NFD.  In any case, indeed this should be mentioned in the readme.
